staging/src/k8s.io/code-generator/cmd/conversion-gen/generators/conversion.go

```diff
@@ -405,29 +315,29 @@ type genConversion struct {
 	// the package that the conversion funcs are going to be output to
 	outputPackage string
 	// packages that contain the peer of types in typesPacakge
-	peerPackages             []string
-	manualConversionsTracker generators.ManualConversionsTracker
-	imports                  namer.ImportTracker
-	types                    []*types.Type
-	explicitConversions      []conversionPair
-	skippedFields            map[*types.Type][]string
-	useUnsafe                TypesEqual
+	peerPackages               []string
+	manualConversionsTracker   *generators.ManualConversionsTracker
+	imports                    namer.ImportTracker
+	types                      []*types.Type
+	explicitConversions        []conversionPair
+	skippedFields              map[*types.Type][]string
+	unsafeConversionArbitrator *generators.UnsafeConversionArbitrator
 }
 
-func NewGenConversion(sanitizedName, typesPackage, outputPackage string, manualConversionsTracker generators.ManualConversionsTracker, peerPkgs []string, useUnsafe TypesEqual) generator.Generator {
+func NewGenConversion(sanitizedName, typesPackage, outputPackage string, manualConversionsTracker *generators.ManualConversionsTracker, peerPkgs []string, unsafeConversionArbitrator *generators.UnsafeConversionArbitrator) generator.Generator {
 	return &genConversion{
 		DefaultGen: generator.DefaultGen{
 			OptionalName: sanitizedName,
 		},
-		typesPackage:             typesPackage,
-		outputPackage:            outputPackage,
-		peerPackages:             peerPkgs,
-		manualConversionsTracker: manualConversionsTracker,
-		imports:                  generator.NewImportTracker(),
-		types:                    []*types.Type{},
-		explicitConversions:      []conversionPair{},
-		skippedFields:            map[*types.Type][]string{},
-		useUnsafe:                useUnsafe,
+		typesPackage:               typesPackage,
+		outputPackage:              outputPackage,
+		peerPackages:               peerPkgs,
+		manualConversionsTracker:   manualConversionsTracker,
+		imports:                    generator.NewImportTracker(),
+		types:                      []*types.Type{},
+		explicitConversions:        []conversionPair{},
+		skippedFields:              map[*types.Type][]string{},
+		unsafeConversionArbitrator: unsafeConversionArbitrator,
```

Comment:
If this can be nil, document that with a function godoc comment (looks like the function needs one anyway) and I suggest (if possible) constructing a default one if the user passes nil.

Reply:
`nil` means unsafe conversions are disabled. Adding a comment.